src/Diagnostics/FxCop/Core/AbstractCodeBlockAnalyzer.cs

```diff
@@ -0,0 +1,22 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System.Collections.Immutable;
+using Microsoft.CodeAnalysis.Diagnostics;
+
+namespace Microsoft.CodeAnalysis.FxCopAnalyzers
+{
+    public abstract class AbstractCodeBlockAnalyzer : DiagnosticAnalyzer
+    {
+        public sealed override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics => ImmutableArray.Create(Descriptor);
```

Comment:
@srivatsn Probably that's just an alternate preference :). I have removed the abstract code block analyzer type I added here.

For me, an analyzer (especially simple stateless ones) is more cleaner if it is declared as one via the type hierarchy (named type analyzer, syntax analyzer, etc.). What I prefer to read is: okay, I am a code block analyzer and I need an AnalyzeCodeBlock method. The option you suggested is that all analyzers are DiagnosticAnalyzer and you need to dig through the Initialize method and registered actions to figure out what it is doing. Later option is good in the sense that everything about the analyzer implementation is in one place, the former option is good in the sense that everything about "my analyzer" implementation is declaratively understood by looking at just the base types and overridden methods :)